### PR TITLE
[DOCS] - Complete Engine docs

### PR DESCRIPTION
**Pull request by dsaltares**

### CODE REVIEW

File: doc/classes/Engine.xml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="Engine" inherits="Object" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Access to basic engine properties.
 	</brief_description>
 	<description>
+		The [Engine] class allows you to query and modify basic parameters regarding the run-time characteristics of your game. For example: frames per second, number of frames rendered so far, time scale, etc.
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: It's not necessary to link a class to itself. 

"parameters regarding" is too wordy, and "basic" is too relative. The last part is a sentence fragment. I would just say:

    allows you to query and modify the game's run-time parameters, such as frames per second, time scale, and others.

File: doc/classes/Engine.xml [hunk 2]
```diff
@@ -41,12 +43,14 @@
 			<return type="float">
 			</return>
 			<description>
+				Return the desired frames per second. If the hardware cannot keep up, this setting may not be respected. It defaults to 0, which indicates no limit.
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: Return -> Returns

File: doc/classes/Engine.xml [hunk 3]
```diff
@@ -41,12 +43,14 @@
...
 			</description>
 		</method>
 		<method name="get_time_scale">
 			<return type="float">
 			</return>
 			<description>
+				Controls how fast or slow the in-game clock ticks versus the real life one. It defaults to 1.0. A value of 2.0 means the game moves twice as fast as real life, whilst a value of 0.5 means the game moves at half the regular speed.
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: The "get" method does not control the time scale, just returns the current value.

File: doc/classes/Engine.xml [hunk 4]
```diff
@@ -67,12 +71,14 @@
...
 			<description>
+				Return whether or not the editor is running.
 			</description>
 		</method>
-		<method name="is_in_fixed_frame" qualifiers="const">
+		<method name="is_in_physics_frame" qualifiers="const">
 			<return type="bool">
 			</return>
 			<description>
+				Return whether or not the game is inside the fixed process and physics phase of the game loop.
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: Per writing guidelines, boolean methods/members prefer "[code]true[/code]" over "whether or not".

  Returns [code]true[/code] if the game is in the fixed process or physics phase of the game loop.

File: doc/classes/Engine.xml [hunk 5]
```diff
@@ -81,6 +87,7 @@
 			<argument index="0" name="enabled" type="bool">
 			</argument>
 			<description>
+				Set whether or not the game is running in the editor.
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: Should use "[code]true[/code]"

File: doc/classes/Engine.xml [hunk 6]
```diff
@@ -98,6 +105,7 @@
 			<argument index="0" name="target_fps" type="int">
 			</argument>
 			<description>
+				Set the target frames per second.
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: Set -> Sets

File: doc/classes/Engine.xml [hunk 7]
```diff
@@ -106,6 +114,7 @@
 			<argument index="0" name="time_scale" type="float">
 			</argument>
 			<description>
+				Set the time scale.
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: Set -> Sets

**Review by cbscribe — CHANGES_REQUESTED** [review]

**Comment by saltares** [comment]
@cbscribe should be ready to go. Thanks.

File: doc/classes/Engine.xml [hunk 8]
```diff
@@ -67,12 +71,14 @@
 			<return type="bool">
 			</return>
 			<description>
+				Returns whether or not the editor is running.
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: Per writing guidelines, boolean methods/members prefer "[code]true[/code]" over "whether or not".

Returns [code]true[/code] if the editor is running.

File: doc/classes/Engine.xml [hunk 9]
```diff
@@ -67,12 +71,14 @@
...
 			</description>
 		</method>
-		<method name="is_in_fixed_frame" qualifiers="const">
+		<method name="is_in_physics_frame" qualifiers="const">
 			<return type="bool">
 			</return>
 			<description>
+				Returns whether or not the game is inside the fixed process and physics phase of the game loop.
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: Per writing guidelines, boolean methods/members prefer "[code]true[/code]" over "whether or not".

Returns [code]true[/code] if the game is in the fixed process or physics phase of the game loop.

File: doc/classes/Engine.xml [hunk 10]
```diff
@@ -89,7 +96,7 @@
 			<argument index="0" name="iterations_per_second" type="int">
 			</argument>
 			<description>
-				Set the amount of fixed iterations per second (for fixed process and physics).
+				Sets the amount of fixed iterations per second (for fixed process and physics).
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: amount -> number

File: doc/classes/Engine.xml [hunk 11]
```diff
@@ -13,7 +15,7 @@
 			<return type="int">
 			</return>
 			<description>
-				Return the total amount of frames drawn.
+				Returns the total amount of frames drawn.
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: amount -> number

File: doc/classes/Engine.xml [hunk 12]
```diff
@@ -27,26 +29,28 @@
 			<return type="int">
 			</return>
 			<description>
-				Return the amount of fixed iterations per second (for fixed process and physics).
+				Returns the amount of fixed iterations per second (for fixed process and physics).
```
Comment (cbscribe) [inline doc/classes/Engine.xml]: amount -> number

**Review by cbscribe — CHANGES_REQUESTED** [review]
Missed a couple of requested changes.

**Comment by saltares** [comment]
Done

**Review by cbscribe — APPROVED** [review]

**Comment by cbscribe** [comment]
Please squash the commits so we can merge.

**Comment by saltares** [comment]
Squashed

**Comment by cbscribe** [comment]
Thanks for your contribution!

**Comment by saltares** [comment]
Thanks for your patience!
